[001-MEMBER/ADMIN-MANAGE] - 회원등록/업데이트/회원List 구현

ikjo39 opened the pull request:
feat - [001-MEMBER/ADMIN-MANAGE]- 회원등록/업데이트/Admin-회원List 구현

로그인 서비스 - 토큰없이 로그인 구현에 미숙하여 아직 구현하지 못하였습니다.

ikjo39 committed:
feat - [001-MEMBER/ADMIN-MANAGE]- 회원등록/업데이트/회원List 구현

로그인 서비스 - 미구현

HeoJeongJu commented:
https://pygmalion0220.tistory.com/entry/Spring-Boot-Rest-API-%EC%97%90-Spring-Security-Form-%EB%A1%9C%EA%B7%B8%EC%9D%B8-%EC%A0%81%EC%9A%A9%ED%95%98%EA%B8%B0

jwt 토큰 없이 구현하는 방법이 있어요!

HeoJeongJu commented on src/main/java/com/ikjo39/commerce/entity/Member.java:
```diff
@@ -0,0 +1,44 @@
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
+	@Column(unique = true)
```
코드리뷰 테스트

spring-kotlin-lover replied:
특정 Table 의 Column 에 unique 설정을 하지 않으면 좋겠네요. email 에 대한 정보가 항상 고유값을 보장한다는 비즈니스로만 이루어질 수 없습니다.

만일 주민번호 기반의 고유성을 가지게 한다고 하면 어떻게 될까요? 설계해주신 엔티티 Table 은 email 에 대한 고유성 제약이 의미없게 되버리기 때문입니다.

따라서 고유성에 대한 비즈니스는 DB 모델로 가져가는 것이 아니라 Service 로직으로 풀면 좋겠네요

ikjo39 replied:
> 특정 Table 의 Column 에 unique 설정을 하지 않으면 좋겠네요. email 에 대한 정보가 항상 고유값을 보장한다는 비즈니스로만 이루어질 수 없습니다.
> 
> 만일 주민번호 기반의 고유성을 가지게 한다고 하면 어떻게 될까요? 설계해주신 엔티티 Table 은 email 에 대한 고유성 제약이 의미없게 되버리기 때문입니다.
> 
> 따라서 고유성에 대한 비즈니스는 DB 모델로 가져가는 것이 아니라 Service 로직으로 풀면 좋겠네요

넵 저는 로직이 변경될 경우를 생각하지 않고 설계를 했던거 같아요
멘토님의 말씀이 더 맞아 프로젝트에 반영해보려합니다
감사드려요!! 😊

spring-kotlin-lover commented on src/main/java/com/ikjo39/commerce/controller/MemberController.java:
```diff
@@ -0,0 +1,53 @@
+
+		return ResponseEntity.ok(member);
+	}
+
```
new line 이 엄청 많네요. 필요없는 부분들은 모두 지워주세요. 

기본적인 코딩 컨벤션은 꼭 지켜주시면 좋겠네요. 코딩 컨벤션에 대해 접해보신적이 없다면 이번 기회에 한번 읽고 접해보시면 좋겠습니다 :)
https://naver.github.io/hackday-conventions-java/

ikjo39 replied:
처음 접하다 보니 기본적인 컨벤션이 있을거란 생각을 못했네요! 공유해주신 게시물 읽어보고 느낀점이 많습니다!! 반영해서 수정해볼게요 
감사합니다!!

spring-kotlin-lover reviewed (commented):
안녕하세요~! 저는 무지멘토라구 하구요. 기본적인 코딩 컨벤션을 한번 봐보시고 코드 전반적으로 적용해보면 좋겠네요 :)

ikjo39 committed:
[#1] Style - [001-MEMBER/ADMIN-MANAGE]- 기본 Code-Convention 적용

ikjo39 committed:
[#1] fix - [001-MEMBER/ADMIN-MANAGE]
     - member, common, admin 패키지 추가 후 정리
     - 로그인 구현의 어려움으로 인해 뒤로 미룸 --> security 주석 추후 Jwt와 Spring Security 이용해서 구현해볼 계획
     - ExceptionHandler 추가
     - RedisConfiguration, SwaggerConfiguration 추가
